rust/linux-client/src/main.rs

```diff
@@ -57,6 +85,23 @@ impl Callbacks for CallbackHandler {
     }
 }
 
+fn get_system_default_resolvers_resolv_conf() -> Result<Vec<IpAddr>> {
```

Comment:
I reckon don't use `anyhow::Error` here. You can use `io::Error` instead and construct an `io::Error::custom` for the `resolv_conf::ParseError`. Or you make yourself a new `Error` enum with both.

Reply:
I could go for a `thiserror` since it's not quite an IO error, but in another PR conectado said to prefer `anyhow` for the client apps, so I did that. I was thinking connlib treats the errors as opaque anyway.

Reply:
> but in another PR conectado said to prefer `anyhow` for the client apps, so I did that

Okay fair!

It does seem a little strange though because you are already declaring an `Error` to workaround `anyhow::Error` not being an `std::error::Error` so it might just be cleaner to use that enum "properly" instead.